### DISCUSSION TITLE
Any plan to release with java >21 support: https://github.com/TNG/ArchUnit/pull/1364/

### DISCUSSION THREAD

**Discussion by robfrank** [TNG/ArchUnit#1394] [General]
As per tittle, is there any plan to do a release that support java > 21? 
This is merged: https://github.com/TNG/ArchUnit/pull/1364/files
Thank you very much for all the work/

**Comment by hankem** [comment]
Thanks for your message! You're right; there should be a new release soon.

**Reply by hankem** [reply]
[ArchUnit 1.4.0](https://github.com/TNG/ArchUnit/releases/tag/v1.4.0) has been released. Sorry for the delay!

**Reply by robfrank** [reply]
Thank you very much!

**Comment by skyblackhawk** [comment]
Hi all,
&nbsp;&nbsp;&nbsp;&nbsp;I discovered this great maven plugin but I have some questions.

1. Does ArchiUnit support WebFlux? We use WebFlux reactive no blocking black pressure mode heavily with the latest version of SpringBoot on Java21 LTS and reactive mongoDB as well as reactive external services in multi threads.
3. Does ArchiUnit have the ability to check the correct hexagonal architecture in the project structure (folder and packages structures, classes, ... )?
4. Can ArchiUnit generate a report in different formats: html, pdf, csv, about the problems found and the solutions to solve them?

Thanks for your reply.
Best regards

**Reply by hankem** [reply]
Hi @skyblackhawk,

your questions seem unrelated to this question #1432 about the release 1.4.0, but would better fit in a new discussion.
I'm nevertheless trying to answer them:

1. I'm not sure which kind of support you're interested in. ArchUnit analyzes relations between classes from byte code; it does not depend or specifically support any framework.
2. Yes, see [§8.1.2. "Onion Architecture" of the ArchUnit user guide](https://www.archunit.org/userguide/html/000_Index.html#_onion_architecture).
3.
    a) ArchUnit does not generate reports itself; it's a library for unit tests. Your test framework may generate reports of failing and passing tests. You can register an [`ArchUnitExtension`](https://javadoc.io/doc/com.tngtech.archunit/archunit/latest/com/tngtech/archunit/lang/extension/ArchUnitExtension.html) that is invoked with the results of all `ArchRule` evaluations, and implement a custom report.
    b) As ArchUnit is very flexible in composing rules to check, it cannot generically provide solutions how to fix violations. But users can provide hints in their rules themselves, see for example the rule [`NO_CLASSES_SHOULD_USE_FIELD_INJECTION`](https://github.com/TNG/ArchUnit/blob/v1.4.0/archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java#L417-L421):
    > no fields should be annotated with an injection annotation, because field injection is considered harmful; use constructor injection or setter injection instead; see https://stackoverflow.com/q/39890849 for detailed explanations

**Comment by skyblackhawk** [comment]
Thanks a lot for your reply.
